[QUERY] KeyVault: Sharelink sample can not be built in vs code

**Environment:**
Sharelink 

I download sharelink.zip and extract it, but build in vs code, error is:
...\ShareLink\Properties\AssemblyInfo.cs(4,23): error CS0234: The type or namespace name 'CodeGenSuppressTypeAttributeAttribute' does not exist in the namespace 'Azure.Core' (are you missing an assembly reference?)

can anyone give me some idea? thanks

I open the VS code from extracted folder；open terminal, run dotnet build:
PS C:\Users\jinnliu\ShareLink> dotnet build

Welcome to .NET Core 3.1!
---------------------
SDK Version: 3.1.409

Telemetry
---------
The .NET Core tools collect usage data in order to help us improve your experience. It is collected by Microsoft and shared with the community. You can opt-out of tlemetry by setting the DOTNET_CLI_TELEMETRY_OPTOUT environment variable to '1' or 'true' using your favorite shell.

Read more about .NET Core CLI Tools telemetry: https://aka.ms/dotnet-cli-telemetry

----------------
Explore documentation: https://aka.ms/dotnet-docs
Report issues and find source on GitHub: https://github.com/dotnet/core
Find out what's new: https://aka.ms/dotnet-whats-new
Learn about the installed HTTPS developer cert: https://aka.ms/aspnet-core-https
Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli-docs
Write your first app: https://aka.ms/first-net-core-app
--------------------------------------------------------------------------------------
Microsoft (R) Build Engine version 16.7.2+b60ddb6f4 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored C:\Users\jinnliu\ShareLink\ShareLink.csproj (in 661 ms).
Properties\AssemblyInfo.cs(4,23): error CS0234: The type or namespace name 'CodeGenSuppressTypeAttributeAttribute' does not exist in the namespace 'Azure.Core' (are
you missing an assembly reference?) [C:\Users\jinnliu\ShareLink\ShareLink.csproj]
Properties\AssemblyInfo.cs(4,23): error CS0234: The type or namespace name 'CodeGenSuppressTypeAttribute' does not exist in the namespace 'Azure.Core' (are you missBuild FAILED.

Properties\AssemblyInfo.cs(4,23): error CS0234: The type or namespace name 'CodeGenSuppressTypeAttributeAttribute' does not exist in the namespace 'Azure.Core' (are 
you missing an assembly reference?) [C:\Users\jinnliu\ShareLink\ShareLink.csproj]
Properties\AssemblyInfo.cs(4,23): error CS0234: The type or namespace name 'CodeGenSuppressTypeAttribute' does not exist in the namespace 'Azure.Core' (are you missing an assembly reference?) [C:\Users\jinnliu\ShareLink\ShareLink.csproj]
    0 Warning(s)
    2 Error(s)

and run dotnet add package Azure.Core, run dotnet build again, error is same.

--- a/sdk/keyvault/samples/sharelink/README.md
+++ b/sdk/keyvault/samples/sharelink/README.md
@@ -14,6 +14,8 @@ extendedZipContent:
   target: /nuget.config
 
 # Shared
+- path: /sdk/core/Azure.Core/src/Shared/AuthorizationChallengeParser.cs
+  target: /Shared/AuthorizationChallengeParser.cs
 - path: /sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs
   target: /Shared/ChallengeBasedAuthenticationPolicy.cs
 ---

--- a/sdk/keyvault/samples/sharelink/ShareLink.csproj
+++ b/sdk/keyvault/samples/sharelink/ShareLink.csproj
@@ -14,6 +14,7 @@
       AZC0100;
     </NoWarn>
     <IncludeAzureCoreSharedCode>true</IncludeAzureCoreSharedCode>
+    <IncludeGeneratorSharedCode>true</IncludeGeneratorSharedCode>
   </PropertyGroup>
 
   <ItemGroup>

--- a/sdk/keyvault/samples/sharelink/ShareLink.csproj
+++ b/sdk/keyvault/samples/sharelink/ShareLink.csproj
@@ -34,7 +34,7 @@
     <PackageReference Update="Azure.Core" Version="1.14.0" />
     <PackageReference Update="Azure.Identity" Version="1.4.0" />
     <PackageReference Update="Azure.Security.KeyVault.Secrets" Version="4.1.1" />
-    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20210510.2" PrivateAssets="All" />
+    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20210602.1" PrivateAssets="All" />
     <PackageReference Update="System.CommandLine" Version="%(VersionOverride)" />
     <PackageReference Update="System.Text.Json" Version="4.6.0" />
   </ItemGroup>

